fs/config/config.go

```diff
@@ -182,32 +255,47 @@ func makeConfigPath() string {
 			}
 		}
 	}
-
-	// If user's configuration directory was found, then try to create it
-	// and assume rclone.conf can be written there. If user supplied config
-	// then skip creating the directory since it will not be used.
-	if cfgpath != "" {
-		// cfgpath != "" implies cfgdir != ""
+	// If we found a configuration directory to be used for new config during search
+	// above, then create it to be ready for rclone.conf file to be written into it
+	// later, and also as a test of permissions to use fallback if not even able to
+	// create the directory.
+	if defaultConfigDir != "" {
+		configFile := filepath.Join(defaultConfigDir, configFileName)
```

Comment:
That should be just `configFile = filepath...` so you carry on using the same global var.